centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -783,43 +812,45 @@ object Operations extends StrictLogging {
-      IO.raiseError(CentaurTestException(s"Metadata doesn't match the expected. Diff: $jsonDiff", workflow, submittedWorkflow))
+      IO.raiseError(CentaurTestException(s"Error during $testType metadata comparison. Diff (changes to mutate from 'expected' to 'actual'): $jsonDiff", workflow, submittedWorkflow))
     }
   }
 
-  def compareJobManagerStyleMetadataBeforeAndAfterArchival(submittedWorkflow: SubmittedWorkflow,
-                                                           workflow: Workflow): Test[Unit] = new Test[Unit] {
+  def assertJobManagerStyleMetadataDidNotChange(testContext: String,
+                                                jmMetadataBeforeArchival: WorkflowMetadata,
+                                                submittedWorkflow: SubmittedWorkflow,
+                                                workflow: Workflow): Test[Unit] = new Test[Unit] {
 
     override def run: IO[Unit] = for {
-      jmMetadataBeforeArchival <- CentaurCromwellClient.metadata(
-        workflow = submittedWorkflow,
-        Option(CentaurCromwellClient.defaultMetadataArgs.getOrElse(Map.empty) ++ jmArgs),
-        archived = Option(false))
       jmMetadataAfterArchival <- CentaurCromwellClient.metadata(
         workflow = submittedWorkflow,
         Option(CentaurCromwellClient.defaultMetadataArgs.getOrElse(Map.empty) ++ jmArgs),
         archived = Option(true))
       jmMetadataObjectBeforeArchival <- IO.fromTry(Try(removeMetadataSourceFromJsObject(jmMetadataBeforeArchival.value.parseJson.asJsObject)))
       jmMetadataObjectAfterArchival <- IO.fromTry(Try(removeMetadataSourceFromJsObject(jmMetadataAfterArchival.value.parseJson.asJsObject)))
-      _ <- validateMetadataJson(jmMetadataObjectBeforeArchival, jmMetadataObjectAfterArchival, submittedWorkflow, workflow)
+      _ <- validateMetadataJson(testType = s"assertJobManagerStyleMetadataDidNotChange ($testContext)", jmMetadataObjectBeforeArchival, jmMetadataObjectAfterArchival, submittedWorkflow, workflow)
     } yield ()
   }
 
-  def validateJobManagerStyleMetadata(submittedWorkflow: SubmittedWorkflow,
-                                      workflow: Workflow,
-                                      originalMetadata: String,
-                                      validateArchived: Option[Boolean] = None): Test[Unit] = new Test[Unit] {
+  def fetchAndValidateJobManagerStyleMetadata(submittedWorkflow: SubmittedWorkflow,
+                                              workflow: Workflow,
+                                              KnownNonSubWorkflowMetadata: Option[String],
+                                              validateArchived: Option[Boolean] = None): Test[WorkflowMetadata] = new Test[WorkflowMetadata] {
 
-    override def run: IO[Unit] = for {
+    def originalMetadataStringIO: IO[String] = KnownNonSubWorkflowMetadata.map(IO.pure)
+      .getOrElse(fetchMetadata(submittedWorkflow, expandSubworkflows = false, validateArchived).map(_.value))
```

Comment:
TOL a `match` might be a bit less mindbendy

Reply:
👍 made this a `match` and added a comment.